src/herder/test/UpgradesTests.cpp

```diff
@@ -2267,16 +2267,16 @@ TEST_CASE("upgrade invalid during ledger close", "[upgrades]")
             *app, makeProtocolVersionUpgrade(
                       Config::CURRENT_LEDGER_PROTOCOL_VERSION - 1)));
     }
+    SECTION("Invalid flags")
+    {
+        // Base Fee / Base Reserve to 0
+        REQUIRE_THROWS(executeUpgrade(*app, makeBaseFeeUpgrade(0)));
+        REQUIRE_THROWS(executeUpgrade(*app, makeBaseReserveUpgrade(0)));
 
-    // Base Fee / Base Reserve to 0
-    REQUIRE_THROWS(executeUpgrade(*app, makeBaseFeeUpgrade(0)));
-    REQUIRE_THROWS(executeUpgrade(*app, makeBaseReserveUpgrade(0)));
-
-    for_versions_to(17, *app, [&] {
```

Comment:
Why remove the versioning here?

Reply:
The versioning breaks the protocol version check that I added in `for_versions` due to the upgrades changing the protocol version. I think the current test is sufficient for what we want to check (upgrade don't work pre-v18, and work for the latest version).

Reply:
Ah, of course. I forgot about that version check, but at least it's helping identify places where we weren't doing what we meant.

I'd definitely feel more comfortable testing every version like we were doing before--I want this PR to be as close to a pure refactor as possible in terms of test coverage. Can you use a similar approach to what you did in the inflation tests, where you called `updateVersion` (although I would use `executeUpgrade` here) before the versioned sections?

Reply:
Done.